psydac/ddm/partition.py

```diff
@@ -132,10 +136,21 @@ def compute_dims( nnodes, gridsizes, min_blocksizes=None, mpi=None, try_uniform=
     return dims, blocksizes
 
 #==============================================================================
-def compute_dims_general( mpi_size, npts ):
+def compute_dims_general( mpi_size, npts, dims_mask=None ):
+
+    if dims_mask is None:
+        dims_mask = [True]*len( npts )
+    else:
+        assert any(dims_mask), 'dims_mask must contain at least one True value.'
```

Comment:
I suggest a couple more checks here:
```suggestion
    if dims_mask is None:
        dims_mask = [True] * len(npts)
    else:
        assert len(dims_mask) == len(npts), "dims_mask must have one entry for each dimension."
        assert all(isinstance(m, bool) for m in dims_mask), "dims_mask must only contain True/False values."
        assert any(dims_mask), "dims_mask must contain at least one True value."
```

Reply:
Agreed. Should I do "commit suggestion"?

Reply:
If you like my suggestion, that is the fastest way to include it.